File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocBlankLinesSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavadocBlankLinesSettings.class)
public interface JavadocBlankLinesSettings {

  Optional<Boolean> afterDescription();

  Optional<Boolean> afterParameterDescriptions();

  Optional<Boolean> afterReturnTag();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocBlankLinesSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc

import com.fasterxml.jackson.annotation.JsonProperty

data class JavadocBlankLinesSettings(

    @JsonProperty("After description")
    val afterDescription: Boolean? = null,

    @JsonProperty("After parameter descriptions")
    val afterParameterDescriptions: Boolean? = null,

    @JsonProperty("After return tag")
    val afterReturnTag: Boolean? = null
)
